quinn/src/connection.rs

```diff
@@ -323,10 +325,23 @@ impl Connection {
     /// Streams are cheap and instantaneous to open unless blocked by flow control. As a
     /// consequence, the peer won't be notified that a stream has been opened until the stream is
     /// actually used.
-    pub fn open_uni(&self) -> OpenUni {
-        OpenUni {
-            conn: self.0.clone(),
-            state: broadcast::State::default(),
+    pub async fn open_uni(&self) -> Result<SendStream, ConnectionError> {
```

Comment:
These hunks of this commit make it hard to review this because seem to include both changing the approach to futures and the new use of `Notify` over our previous `Broadcast` thing. Do you think it might be feasible to separate two in two steps?

Reply:
Good thought; split out the `async fn` stuff into a separate commit.

Reply:
Felt inspired and split up the refactoring some more as well.